python/mxnet/optimizer.py

```diff
@@ -383,9 +383,9 @@ def create_state(self, index, weight):
     def update(self, index, weight, grad, state):
         assert(isinstance(weight, NDArray))
         assert(isinstance(grad, NDArray))
+        self._update_count(index)
```

Comment:
?

Reply:
If self._update_count(index) was located after self._get_lr(index), the parameter 'self.num_update' used in the function self._get_lr(index) will like bellow.
**INFO:root:index: 0    num_update: 0**
INFO:root:index: 1    num_update: 1
INFO:root:index: 2    num_update: 1
INFO:root:index: 3    num_update: 1
INFO:root:index: 4    num_update: 1
INFO:root:index: 5    num_update: 1
INFO:root:index: 6    num_update: 1
INFO:root:index: 7    num_update: 1
**INFO:root:index: 0    num_update: 1**
INFO:root:index: 1    num_update: 2
INFO:root:index: 2    num_update: 2
INFO:root:index: 3    num_update: 2
INFO:root:index: 4    num_update: 2
INFO:root:index: 5    num_update: 2
INFO:root:index: 6    num_update: 2
INFO:root:index: 7    num_update: 2
**INFO:root:index: 0    num_update: 2**
INFO:root:index: 1    num_update: 3
INFO:root:index: 2    num_update: 3
INFO:root:index: 3    num_update: 3
INFO:root:index: 4    num_update: 3
INFO:root:index: 5    num_update: 3
INFO:root:index: 6    num_update: 3
INFO:root:index: 7    num_update: 3

In the first time update, the 'self.num_update' used in index_0 is 0 and 1 for other index. The same wrong will occur in the next update loop. 
So when lr for indexes except index_0 changed in the function self.lr_scheduler, the index_0 will get the new lr in the next loop.
I move self._update_count before self.get_lr. It can make the lr_scheduler correctly.